### PR TITLE
Pricing support for Admin API & SDK

### PR DESCRIPTION
**Pull request by damianlegawiec**
> [!NOTE]
> **Medium Risk**
> Bulk price upserts bypass model callbacks and touch catalog pricing paths; incorrect rows or scope bugs could corrupt store prices at scale.
> 
> **Overview**
> Adds **admin pricing** end-to-end: **price lists** (CRUD, activate/deactivate, rule-type discovery, membership and nested rules/prices on a single PATCH), **prices** (CRUD plus `bulk_upsert` / `bulk_destroy` keyed on variant/currency/price list), and **read-only markets** for rule pickers.
> 
> The **Admin API** wires store-scoped controllers and serializers (`PriceList`, `PriceRule`, admin `Price`), with bulk writes delegated to `Spree::Prices::BulkUpsert` (fast path that skips normal price callbacks). **`@spree/admin-sdk`** exposes `markets`, `priceLists`, and `prices` with typed params and examples.
> 
> **`CLAUDE.md`** is tightened with explicit backend rules (logic in models/services, Yard on public APIs, no controller/serializer business logic) and test rules (factories only, controller specs for edge cases vs brittle integration tests).
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 5bec5bbcb9781299cdead911df4adc03a802765c. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

## Summary by CodeRabbit

* **New Features**
  * Price lists: full CRUD, activate/deactivate, rule types, price export/listing, spreadsheet-style price rows.
  * Prices: CRUD plus bulk upsert and bulk destroy with locale-aware parsing and reconciliation.
  * Markets: read-only list/get API.
  * Admin SDK: added markets/price-lists/prices endpoints and many TypeScript examples demonstrating usage.

* **Core**
  * Typed-association support, store-scoped prices, permission updates, and model plumbing for price lists/prices.

* **Documentation**
  * Strengthened architecture and test guidelines.

* **Tests**
  * Extensive controller, integration, and service coverage for new APIs and bulk operations.

[Review Change Stack](https://app.coderabbit.ai/change-stack/spree/spree/pull/14085?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

File: spree/api/app/serializers/spree/api/v3/admin/price_list_serializer.rb [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+          # Cheap counts so the index can render "12 products / 36 prices"
+          # without forcing each row to expand the children.
+          attribute :products_count do |pl|
+            pl.products.count
+          end
+
+          attribute :prices_count do |pl|
+            pl.prices.count
+          end
```
[the remark this answers is not in the document]
Comment (damianlegawiec) [reply spree/api/app/serializers/spree/api/v3/admin/price_list_serializer.rb:37]: this is fine, it's not an issue
